lang/cs/Org.Apache.REEF.IMRU/OnREEF/Driver/IMRUApplicationException.cs

```diff
@@ -0,0 +1,57 @@
+﻿// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+using System;
+using Org.Apache.REEF.Utilities.Logging;
+
+namespace Org.Apache.REEF.IMRU.OnREEF.Driver
+{
+    /// <summary>
+    /// Type of exception thrown when possible bugs are detected in IMRU code.
+    /// For example, we reach forbidden region of codes, inconsistent state etc.
+    /// </summary>
+    public sealed class IMRUApplicationException : Exception
```

Comment:
The name is misleading. When I saw the type without the documentation, I assumed this would refer to a bug in an IMRU application, not IMRU itself. Maybe call this `IMRUSystemException`?